src/messageObjects/message.tsx

```diff
@@ -12,8 +12,24 @@ export class HighlightMessage implements Message {
     action: string = "highlightElement";
 }
 
+export class HighlightAndRemovePreviousMessage implements Message {
+    constructor(public newElement: ElementObject, public previousElement: ElementObject) {
+        this.newElement = newElement;
+        this.previousElement = previousElement;
+    }
+    action: string = "highlightAndRemovePrevious";
+}
+
+export class UnhighlightAllAndHighlightSingleMessage implements Message {
+    constructor(public element: ElementObject, public elementType: ElementType) {
+        this.element = element;
+        this.elementType = elementType;
+    }
+    action: string = "unhighlightAllAndHighlightSingle";
+}
+
 export class HighlightAllMessage implements Message {
-    constructor(public type: ElementType, public isChecked: boolean){
+    constructor(public type: ElementType, public isChecked: boolean) {
         this.type = type;
         this.isChecked = isChecked;
     }
```

Comment:
Do you need to create a new class for every message?

Reply:
Each new class is made to handle different payload. To make sure that each case only checks for the right elements.

An alternative would be to just have a field payload, but then that payload would have to take in alot of different types.
Might be other ways to do it, but i thought this was an easy and safe way to do it